### PR TITLE
ESQL: LOOKUP JOIN security tests and privileges

### PR DESCRIPTION
**Pull request by ivancea**
- Remove Enrich privilege for LOOKUP JOIN
- Add security tests to lookup, to ensure they work/fail depending on user roles and privileges

### CODE REVIEW

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java [hunk 1]
```diff
@@ -143,7 +144,7 @@ abstract class AbstractLookupService<R extends AbstractLookupService.Request, T
 
     AbstractLookupService(
         String actionName,
-        String privilegeName,
+        @Nullable String privilegeName,
```
Comment (ivancea) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java:147]: This is a very plain change to make it work, but I think we don't need much more for now. Easy to change later if we want something more complex

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java [hunk 2]
```diff
@@ -239,7 +240,9 @@ public final void lookupAsync(R request, CancellableTask parentTask, ActionListe
 
     private void hasPrivilege(ActionListener<Void> outListener) {
         final Settings settings = clusterService.getSettings();
-        if (settings.hasValue(XPackSettings.SECURITY_ENABLED.getKey()) == false || XPackSettings.SECURITY_ENABLED.get(settings) == false) {
+        if (privilegeName == null
```
Comment (craigtaverner) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java:243]: Perhaps we should have a comment here explaining why we have null and what it means? Alternatively we could move this check to an abstract method and implement differently for ENRICH and LOOKUP JOIN.

**Review by craigtaverner — APPROVED** [review]
LGTM. Did have a comment, but not something that needs to block the PR.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java (hunk 2 above)
Comment (ivancea) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java:249]: Made a simple abstract "getRequiredPrivilege()" method with the javadoc explaining it for null

**Pull request merged**